string product_code
string status
string message